tests/framework/db/sqlite/ConnectionTest.php

```diff
@@ -90,6 +96,41 @@ public function testMasterSlave()
         }
     }
 
+    public function testMasterSlaveRandomization()
```

Comment:
this test looks to complex and will be hard  to maintain in the future. Cant you refactor it to less code?